### PR TITLE
Upgrade to support hyper1

### LINKED ISSUE #13: Update to hyper 1
@ark0f would you accept contributions?

### PR DESCRIPTION
**Pull request by aviramha**
Closes #13

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,24 +16,27 @@ edition = "2018"
...
+hyper = { version = "1", features = ["server", "http1", "client"] }
 async-socks5 = "0.5"
 futures = "0.3"
 tokio = "1.0"
 thiserror = "1.0"
-http = "0.2"
-
+http = "1"
+tower-service = "0.3"
+hyper-util = { version = "0.1", features = ["tokio"]}
```
Comment (ark0f) [inline Cargo.toml]:
```suggestion
hyper-util = { version = "0.1", features = ["tokio"] }
```

File: src/lib.rs [hunk 2]
```diff
@@ -160,7 +162,7 @@ where
             .call(self.proxy_addr)
             .await
             .map_err(Into::<BoxedError>::into)?;
-        let mut buf_stream = BufStream::new(stream); // fixes issue #3
+        let mut buf_stream = TokioIo::new(stream); // fixes issue #3
```
Comment (ark0f) [inline src/lib.rs]:
```suggestion
        let mut buf_stream = BufStream::new(TokioIo::new(stream)); // fixes issue #3
```

File: Cargo.toml [hunk 3]
```diff
@@ -16,24 +16,27 @@ edition = "2018"
 github-actions = { repository = "https://github.com/ark0f/hyper-socks2", workflow = "CI" }
 
 [dependencies]
-hyper = "0.14"
+hyper = { version = "1", features = ["server", "http1", "client"] }
```
Comment (ark0f) [inline Cargo.toml]: It looks like we don't need any of these features

File: Cargo.toml [hunk 4]
```diff
@@ -16,24 +16,27 @@ edition = "2018"
...
-rusttls = { package = "rustls", version = "0.21", optional = true }
-rustls-native-certs = { version = "0.6", optional = true }
+hyper-rustls = { version = "0.26", optional = true }
+rusttls = { package = "rustls", version = "0.22", optional = true }
+rustls-native-certs = { version = "0.7", optional = true }
 
 [dev-dependencies]
-hyper = { version = "0.14", features = ["http1"] }
 tokio = { version = "1.0", features = ["macros"] }
+hyper-util = { version = "0.1", features = ["http1", "client", "client-legacy"]}
```
Comment (ark0f) [inline Cargo.toml]:
```suggestion
hyper-util = { version = "0.1", features = ["http1", "client", "client-legacy"] }
```

**Review by ark0f — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (aviramha) [reply Cargo.toml]: Done

File: src/lib.rs (hunk 2 above)
Comment (aviramha) [reply src/lib.rs:165]: this leads to
```
mismatched types
expected associated type `<C as tower_service::Service<Uri>>::Response`
            found struct `TokioIo<<C as tower_service::Service<Uri>>::Response>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B4%5D?4#file%3A%2F%2F%2FUsers%2Faviramhassan%2FCode%2Fhyper-socks2%2Fsrc%2Flib.rs)
lib.rs(168, 9): arguments to this enum variant are incorrect
lib.rs(168, 9): the type constructed contains `TokioIo<<C as tower_service::Service<Uri>>::Response>` due to the type of the argument passed
result.rs(506, 5): tuple variant defined here
```

**Comment by aviramha** [comment]
Thanks for the review. I think I resolved all comments.

**Review by ark0f — APPROVED** [review]
Thank you for contribution

**Pull request merged**